hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/adam/AdamJob.java

```diff
@@ -138,9 +138,19 @@ protected boolean setupJob(DistributedFileSystemOps dfso, YarnClient yarnClient)
     runnerbuilder.setJobName(jobconfig.getAppName());
 
     try {
+
+      String usersName = "";
```

Comment:
I suggest `usersName` is initialized to null. if `fname` is empty or null and `lname` is not, the name will end up with a whitespace in the beginning. you can do usersName.trim() if usersName is not null

Reply:
It will be problematic if there is no Fname or Lname, then it will try to export an environment variable with value null. Which would either lead to "null" being the name or crashing during startup.

Reply:
If it's null what's the point of exporting it? We can check in SparkYarnRunnerBuilder if it's null and if not export it. Would that work?

Reply:
Okay would work, then I will also have to do a fix in hops-util-py. On it.